vnext/Microsoft.ReactNative/IReactDispatcher.cpp

```diff
@@ -17,7 +19,16 @@ bool ReactDispatcher::HasThreadAccess() noexcept {
 }
 
 void ReactDispatcher::Post(ReactDispatcherCallback const &callback) noexcept {
-  return m_queue.Post([callback]() noexcept { callback(); });
+  return m_queue.Post([callback]() noexcept {
+    try {
+      callback();
+    } catch (winrt::hresult_error const &error) {
+      std::stringstream errorCode;
+      errorCode << "0x" << std::hex << error.code();
+      LOG(ERROR) << "HRESULT " << errorCode.str() << ": " << winrt::to_string(error.message());
+      throw;
```

Comment:
>throw [](http://example.com/codeflow?start=6&length=5)

We probbaly must use the std::terminate here to make compiler happy.

Reply:
std::terminate will get called because it's noexcept, at which point the terminate handler is called, which will call the code that shows the messagebox i added a couple of months ago

Reply:
@asklar - do you have a reference to the PR that added the message box? Perhaps its better for us to wire up behavior in the messagebox to add details to our logs.

Reply:
Found it: https://github.com/microsoft/react-native-windows/commit/141d453df621de42e8de542fce3a7ad23f098c64

Perhaps we just add the log there instead...

Reply:
On second thought, I don't think the existing termination guard is a good spot for this. This fix ensures that if the user does appropriately context switch their native method onto a dispatcher (whether JS, UI, or otherwise) and the operation fails, there will be some opportunity to get logs. We could take the same approach as you do for the terminate handler, but not sure that's needed.